botocore/client.py

```diff
@@ -208,6 +209,10 @@ def _get_client_args(self, service_model, region_name, is_secure,
         self._inject_s3_configuration(
             config_kwargs, scoped_config, client_config)
         self._conditionally_unregister_fix_s3_host(endpoint_url, event_emitter)
+        if service_name == 's3':
+            # We want to make double sure that we don't create this object
```

Comment:
We only need this for S3, and at this point we have a trivially easy way to limit it to S3. Otherwise we have that object laying around taking up memory and hanging on to the endpoint bridge for services that don't need it.